File: Random/app/src/main/java/com/example/random/MainActivity.java
package com.example.random;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.SeekBar;
import android.widget.TextView;

import java.util.Random;

public class MainActivity extends AppCompatActivity {

    private Button btnGenerate;
    private TextView txResult;
    private SeekBar seekBar;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btnGenerate = findViewById(R.id.button);
        txResult = findViewById(R.id.textResult);
        seekBar = findViewById(R.id.seekBar);

        btnGenerate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Random rand = new Random();
                int num = rand.nextInt(seekBar.getProgress());
                txResult.setText(Integer.toString(num));
            }
        });
    }
}

File: KotlinGenerateNumber/app/src/main/java/com/example/kotlingeneratenumber/MainActivity.kt
package com.example.kotlingeneratenumber

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnGenerate.setOnClickListener{
            val rand = Random();
            textView.text = rand.nextInt(seekBar.progress).toString();
        }
    }
}
